### PR TITLE
Open up CORS for everyone

### PR DESCRIPTION
**Pull request by JulianKniephoff**
I think it would be nice if people could use the test servers to test whatever Opencast backed web app they are developing.

### CODE REVIEW

**Comment by gregorydlogan** [comment]
This seems reasonable enough to me, although I'm not 100% familiar with the ramifications.  Worst case we end up with a (temporarily) broken demo box, right?

**Comment by JulianKniephoff** [comment]
That could happen if I broke the config, yes. Let's hope I didn't. x)

If you are (or anyone else is, for that matter) is wondering about the security implications of this: CORS is a pure client side protection. These headers signal to the client (e.g. a browser) whether or not a request it is trying to make is allowed or not. There are of course clients which ignore these headers, and this is actually the norm, outside of the browser realm.

That means an attacker can make any request they want, even with the current setup. They **can't** do it making the user's browser execute some script they injected into any random website, highjacking their session or something like that. They would have to inject code specifically into `studio.opencast.org`.

So with this change, an attacker could XSS-attack a user on any website, making requests to our test servers in their name. Now the question is: Why would you ever do that, if our web server configuration also autofills the login form with valid admin credentials. :stuck_out_tongue_winking_eye: 

TL;DR: The security implications are near 0, I'd say. #famouslastwords

**Comment by lkiesow** [comment]
The security implications are that users might use the same configuration for their own servers since this is a known working and well-maintained configuration. So there are no concerns regarding the demo servers at all. But there are concerns regarding the ecosystem. That is the reason for the security-related Nginx configuration being rather strict in the first place.

So, the question becomes:

- do we want to open this up globally, saying that browsers should generally allow scripts to interact with the demo servers from other websites? *(← that's what this pull request does)*
- do we want to keep the current strict setting where it is allowed for studio.opencast.org only?
- do we want to additionally allow specific other systems/domains (e.g. localhost) but not generally allow this?

@gregorydlogan, any thoughts?

I'm hesitant to generally open this but could live with that if it significantly helps developers.

**Comment by gregorydlogan** [comment]
I feel like opening it up to localhost probably enables 90% of the development scenarios I can think of, the other 10% you should really stand your own demo box up for.  That being said, I don't actually care if we open it up globally either.

@JulianKniephoff, would opening up localhost be enough to satisfy your usecase?

**Comment by JulianKniephoff** [comment]
My specific, current use case would be covered by allowing acces from `localhost`. But my specific use case would for example also see the content security policy and frame options altered and/or removed. :thinking: However, a few weeks ago, I wanted to share the current state of my project with the team, for example, so I temporarily served it under some domain name. That would not have worked, then. (It actually **did** work IIRC, because @lkiesow temporarily opened up CORS globally at some point?)

And of course neither would demoing an app that uses the demo servers to the rest of the world; `studio.opencast.org` gets to do this, because we decided to integrate it, but in general, every Opencast based webapp would have to set up their own demo server and fill it with content, or do some proxy trickery, in which case, these requests end up on the community server again anyway.

But okay, if we say that this configuration is supposed to be a good example of a secure setup, I think we should drop this PR, because I would need to open up even more to get it to work in my case. However, IMO the goals of providing a useful, comprehensive demo server, and making it a model for good security practices are at odds, cf. filling in admin credentials automatically. And I think the latter purpose would be better served by a separate project. Is this even advertised as such anywhere? Because I for one would not look for that in a demo server. :thinking:

**Comment by gregorydlogan** [comment]
What about a happy medium: Let's make this a parameter based on the hostvars, and have it default to off.  That way we can leave the demo servers wide open, but if someone is going to fork this as a known working version they would have to also copy the hostvars.

**Comment by lkiesow** [comment]
pragmatic solution: just merge this.
